test/sql/function/list/list_select.test

```diff
@@ -0,0 +1,254 @@
+# name: test/sql/function/list/list_select.test
+# description: Test list_select function
+# group: [list]
```

Comment:
Let's add `PRAGMA enable_verification`.

Reply:
Why do I get an `Invalid PhysicalType for GetTypeIdSize` for `[INTERVAL 1 YEAR]` when I enable verification? See FIXME.

Reply:
Nice catch; this is a bug on our side! I'll ping you once I've got a PR up (it should be this week). Then you can merge once it's on the main branch.

This query (on master) also throws the same internal error.
```sql
statement error
SELECT list_concat(list_value(to_years(1)), CAST(list_value(1) AS BIGINT[]));
----
Conversion Error

statement ok
PREPARE s1 AS SELECT list_concat(list_value(to_years($1)), CAST(list_value($1) AS BIGINT[]));

statement ok
EXECUTE s1(1);

================================================================================
EXECUTE s1(1);
================================================================================
INTERNAL Error: Invalid PhysicalType for GetTypeIdSize
```